invenio_rdm_records/services/components/pids.py

```diff
@@ -244,10 +243,27 @@ def edit(self, identity, draft=None, record=None):
         # FIXME: this should be moved to marshmallow level
         # validation happens before to be able to fail before reserving pids
         # it means a double looping but validation should eventually be moved
-        # to marshmellow see invenio-rdm-records/issues/796
-        errors = {}
+        # to marshmallow see invenio-rdm-records/issues/796
+        errors = []
         self._validate_pids_schemes(record_pids)
         self._validate_pids(record_pids, record, errors)
         if errors:
             raise ValidationError(message=errors)
         draft.pids = record_pids
+
+    def post_publish(self, identity, record=None):
+        """Post publish handler."""
+        service = current_rdm_records.records_service
```

Comment:
my bad, I overlooked this. The `components` have the `service` in them (i.e. `self.service`) so there is no need to access it through the proxy. See [here](https://github.com/inveniosoftware/invenio-records-resources/blob/master/invenio_records_resources/services/records/components.py#L22)